hal/src/main/native/include/HAL/Types.h

```diff
@@ -48,3 +48,9 @@ typedef HAL_Handle HAL_CANHandle;
 typedef HAL_CANHandle HAL_PDPHandle;
 
 typedef int32_t HAL_Bool;
+
+#ifdef __cplusplus
+#define HAL_ENUM(name) enum name : int32_t
+#else
+#define HAL_ENUM(name) typedef int32_t name; enum name
```

Comment:
Typedefing `int32_t` like this doesn't seem to have any effect. Changing this to `int16_t` still results in `sizeof(enum name) = 4` on my machine.
```[tav@myriad ~]$ cat Main.c
#include <stdint.h>
#include <stdio.h>

typedef int16_t name; enum name { kOne, kTwo };

int main() {
  printf("sizeof(enum name) = %i\n", sizeof(enum name));
  return 0;
}
[tav@myriad ~]$ gcc -o Main Main.c
[tav@myriad ~]$ ./Main
sizeof(enum name) = 4
```

Reply:
I think this will be fine for most purposes. The idea is that the bare enum name is the correct sized int type for the function signatures.

Reply:
There's still going to be truncation if the user passes in an actual `enum name` to the argument, but I'm not sure how that can be helped.

Reply:
Yeah the API is most important. We don't have big enums, we just do need them to be fixed size.